mocks/auth_handler_mock.go

```diff
@@ -0,0 +1,34 @@
+package mocks
+
+import (
+	"net/http"
+
+	"github.com/ONSdigital/dp-authorisation/auth"
+)
+
+type PermissionCheckCalls struct {
+	Calls int
+}
+
+type AuthHandlerMock struct {
+	Required *PermissionCheckCalls
+}
+
+func NewAuthHandlerMock() *AuthHandlerMock {
+	return &AuthHandlerMock{
+		Required: &PermissionCheckCalls{
+			Calls: 0,
+		},
+	}
+}
+
+func (a AuthHandlerMock) Require(required auth.Permissions, handler http.HandlerFunc) http.HandlerFunc {
+	return a.Required.checkPermissions(handler)
+}
+
+func (c *PermissionCheckCalls) checkPermissions(h http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		c.Calls += 1
+		h.ServeHTTP(w, r)
```

Comment:
You may need to be careful here if tests run in parallel you have a potential race condition. Consider adding a sync.lock around the var.

Reply:
Hmmm ...
I copied that code from dp-dataset-api ... and i think its still like that ...

Added mutex, etc. in dp-topic-api